pkg/service/admission/externalip_admission.go

```diff
@@ -84,11 +86,27 @@ func (r *externalIPRanger) Admit(a kadmission.Attributes) error {
 		return nil
 	}
 
+	// Determine if an ingress ip address should be allowed as an
+	// external ip by checking the loadbalancer status of the previous
+	// object state.
+	ingressIP := ""
+	if r.allowIngressIP && svc.Spec.Type == kapi.ServiceTypeLoadBalancer {
+		old, ok := a.GetOldObject().(*kapi.Service)
```

Comment:
check both ok and old != nil, since someone can pass a nil api.Service pointer that would be ok = true